packages/compile-service/src/index.ts

```diff
@@ -33,50 +33,41 @@ async function findTransactionWithRetry(
     tx = await modelTransaction.findById(id);
 
     if (tx) {
-      break;
+      return tx;
     }
 
     await setTimeout(intervalMs);
   }
 
-  if (!tx) {
-    logger.error(
-      `Transaction ${id} has not been found after ${maxWaitTimeMs} ms`
-    );
-  }
-
-  return tx;
+  logger.error(
+    `Transaction ${id} has not been found after ${maxWaitTimeMs} ms`
+  );
+  return null;
 }
 
-(async () => {
-  // Init zkAppCompiler
+async function processQueue(redisQueue: RedisQueueService<DbTransactionQueue>) {
   const zkAppCompiler = new ZkCompileService({
     networkId: config.NETWORK_ID,
     mina: config.MINA_URL,
   });
-  // Init redis queue service
-  const redisQueue = new RedisQueueService<DbTransactionQueue>(
-    "zkAppDeploymentQueue",
-    { url: config.REDIS_URL }
-  );
-  // Connect to db
+
   const serviceDb = DatabaseEngine.getInstance(config.MONGODB_URL);
   const proofDb = DatabaseEngine.getInstance(config.PROOF_MONGODB_URL);
 
-  if (!serviceDb.isConnected()) {
-    await serviceDb.connect();
-  }
-
-  if (!proofDb.isConnected()) {
-    await proofDb.connect();
-  }
+  if (!serviceDb.isConnected()) await serviceDb.connect();
+  if (!proofDb.isConnected()) await proofDb.connect();
 
   const modelTransaction = ModelDbTransaction.getInstance();
   const modelDbSettings = ModelDbSetting.getInstance();
 
   while (true) {
-    const request = await redisQueue.dequeue();
-    if (request) {
+    try {
+      const request = await redisQueue.dequeue();
+      if (!request) {
+        await setTimeout(1000); // Prevent busy looping when the queue is empty
```

Comment:
I think you should use global variable for timeout

Reply:
done